### DISCUSSION TITLE
is there any method or attribute which is used to prevent the extraction of watermark, page number or background text.

### DISCUSSION THREAD

**Discussion by hemilparmar** [pymupdf/PyMuPDF#1884] [Q&A] [answered]
when i try to extract some unwanted text is also coming so want to ask you do pymupdf has any method or attribute to prevent those kind of extraction. please answer this.

**Comment by JorjMcKie** [comment, marked as answer]
moving this to "Discussions" first

**Reply by JorjMcKie** [reply]
You must specify criteria for any text you do not want to consider.
If you use `get_text("dict" <any parameters>)` you will get all available information to make decision: text, color, text box, writing direction, font name, font size.
The relative sequence of the text items extracted also indicates the sequence in which the items were stored. This could e.g. be used to qualify text as "background": if there are text pieces coming later overlapping its position.
